### PR TITLE
CDS-87266-Added all details reg the Approval step in Platform docs and added the relevant link in the same topic in CD docs

### PR DESCRIPTION
**Pull request by roshnisarangadharan**
Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

https://harness.atlassian.net/browse/CDS-87266

### CODE REVIEW

File: docs/continuous-delivery/x-platform-cd-features/cd-steps/approvals/using-harness-approval-steps-in-cd-stages.md [hunk 1]
```diff
@@ -12,272 +12,71 @@ Approvals are usually added in between stage steps to prevent the stage executio
...
-- [Policy Enforcement](/docs/platform/governance/policy-as-code/harness-governance-overview)
-
-## Approval variables
-
-After an approval is granted, [approval expressions](/docs/platform/variables-and-expressions/harness-variables.md#approval-expressions) store the approver name and email as well as any approval comments. These variables are useful if you want the pipeline to generate notifications about the approval. These expressions are not available for all approval types.
 
 ## See also
 
-- [Update Jira Issues in CD stages](/docs/continuous-delivery/x-platform-cd-features/cd-steps/ticketing-systems/update-jira-issues-in-cd-stages)
+- [Using Manual Harness Approval Steps in CD Stages](/docs/continuous-delivery/x-platform-cd-features/cd-steps/approvals/using-harness-approval-steps-in-cd-stages/)
```
Comment (naman-goenka) [inline docs/continuous-delivery/x-platform-cd-features/cd-steps/approvals/using-harness-approval-steps-in-cd-stages.md:79]: isn't this a self-link?

Comment (roshnisarangadharan) [inline docs/continuous-delivery/x-platform-cd-features/cd-steps/approvals/using-harness-approval-steps-in-cd-stages.md:79]:
```suggestion

```

Comment (roshnisarangadharan) [reply docs/continuous-delivery/x-platform-cd-features/cd-steps/approvals/using-harness-approval-steps-in-cd-stages.md:79]: Actually we don't want the platform link here, as it is already mentioned in the top.

**Review by abhinavmittal-harness — APPROVED** [review]

File: docs/platform/approvals/adding-harness-approval-stages.md [hunk 2]
```diff
@@ -104,25 +119,158 @@ Enable this option to reject old executions waiting for approval when a latest s
...
+</details>
 
 ### Prevent approval by pipeline executor
 
 If you don't want the User that initiated the Pipeline execution to approve this step, select the **Disallow the executor from approving the pipeline** option.
 
 ### Approver inputs
 
-You can enter variables and when the approver views the step they can provide new values for the variables.
+In **Inputs to be provided by approver**, you can enter variables and when the approver views the step they can provide new values for the variables.
```
Comment (roshnisarangadharan) [inline docs/platform/approvals/adding-harness-approval-stages.md:273]:
```suggestion
In **Approver Inputs**, you can enter variables and when the approver views the step they can provide new values for the variables.
```

[the remark this answers is not in the document]
Comment (roshnisarangadharan) [reply docs/platform/approvals/adding-harness-approval-stages.md]: Done

File: docs/continuous-delivery/x-platform-cd-features/cd-steps/approvals/using-harness-approval-steps-in-cd-stages.md [hunk 3]
```diff
@@ -12,272 +12,51 @@ Approvals are usually added in between stage steps to prevent the stage executio
 
 For example, in a [Kubernetes blue green deployment](/docs/continuous-delivery/deploy-srv-diff-platforms/kubernetes/kubernetes-executions/create-a-kubernetes-blue-green-deployment), you might want to add an approval step between the stage deployment step, where the new app version is deployed to the staging environment, and the Swap Primary with Stage step, where production traffic is routed to the pods for the new version.
 
-Other approval methods are:
-
-- [Harness Approval stages](/docs/platform/approvals/adding-harness-approval-stages): Add Approval stages for manual intervention.
-- [Adding Jira Approval stages and steps](/docs/platform/approvals/adding-jira-approval-stages): Add Jira Approval stages and steps.
-- [Adding ServiceNow Approval steps and stages](/docs/platform/approvals/service-now-approvals): Add ServiceNow Approval stages and steps.
```
Comment (naman-goenka) [inline docs/continuous-delivery/x-platform-cd-features/cd-steps/approvals/using-harness-approval-steps-in-cd-stages.md:19]: We can add snow cross linking in the second main document : docs/platform/approvals/adding-harness-approval-stages.md

File: docs/platform/approvals/adding-harness-approval-stages.md [hunk 4]
```diff
@@ -104,25 +119,158 @@ Enable this option to reject old executions waiting for approval when a latest s
...
+            description: ""
+            required: false
+            value: 2023-12-20 09:24 AM
+```
+
+</details>
 
 ### Prevent approval by pipeline executor
 
 If you don't want the User that initiated the Pipeline execution to approve this step, select the **Disallow the executor from approving the pipeline** option.
```
Comment (naman-goenka) [inline docs/platform/approvals/adding-harness-approval-stages.md:270]: We can add `Even if the User is in the user group selected in User Group, they won't be able to approve this step.`

Comment (naman-goenka) [reply docs/platform/approvals/adding-harness-approval-stages.md:270]: (was in CD doc)

File: docs/platform/approvals/adding-harness-approval-stages.md [hunk 5]
```diff
@@ -84,7 +88,18 @@ Click the **Approval** step.
```
Comment (naman-goenka) [inline docs/platform/approvals/adding-harness-approval-stages.md:88]: we can add `isAutoRejectEnabled: false` : techdebt (is present in default yaml generated via FE)
https://qa.harness.io/ng/account/rXUXvbFqRr2XwcjBu3Oq-Q/home/orgs/default/projects/testNaman/pipelines/HAApp/pipeline-studio/?storeType=INLINE&stageId=dsdsa&sectionId=EXECUTION

File: docs/platform/approvals/adding-harness-approval-stages.md [hunk 6]
```diff
@@ -104,25 +119,158 @@ Enable this option to reject old executions waiting for approval when a latest s
...
+            description: ""
+            required: false
+            value: 2023-12-20 09:24 AM
+```
+
+</details>
 
 ### Prevent approval by pipeline executor
 
 If you don't want the User that initiated the Pipeline execution to approve this step, select the **Disallow the executor from approving the pipeline** option.
```
Comment (roshnisarangadharan) [inline docs/platform/approvals/adding-harness-approval-stages.md:269]:
```suggestion
If you don't want the User that initiated the Pipeline execution to approve this step, select the **Disallow the executor from approving the pipeline** option. Even if the User is in the selected in User Group, they won't be able to approve this step.
```

File: docs/continuous-delivery/x-platform-cd-features/cd-steps/approvals/using-harness-approval-steps-in-cd-stages.md [hunk 7]
```diff
@@ -12,272 +12,51 @@ Approvals are usually added in between stage steps to prevent the stage executio
...
+           - name: foo
+             defaultValue: bar
+   ```
 
 ## See also
 
-- [Update Jira Issues in CD stages](/docs/continuous-delivery/x-platform-cd-features/cd-steps/ticketing-systems/update-jira-issues-in-cd-stages)
+- [Update Jira Issues in CD Stages](/docs/continuous-delivery/x-platform-cd-features/cd-steps/ticketing-systems/update-jira-issues-in-cd-stages)
+- [Using Harness Approval APIs](/kb/continuous-delivery/articles/harness-approval-api)
+
```
Comment (roshnisarangadharan) [inline docs/continuous-delivery/x-platform-cd-features/cd-steps/approvals/using-harness-approval-steps-in-cd-stages.md:62]:
```suggestion
- [Add ServiceNow Approval steps and stages](/docs/platform/approvals/service-now-approvals)
```

File: docs/continuous-delivery/x-platform-cd-features/cd-steps/approvals/using-harness-approval-steps-in-cd-stages.md (hunk 3 above)
Comment (roshnisarangadharan) [reply docs/continuous-delivery/x-platform-cd-features/cd-steps/approvals/using-harness-approval-steps-in-cd-stages.md:19]: This is deletes section. Have added it in the doc now.

**Review by naman-goenka — APPROVED** [review]

**Pull request merged**